src/base/Element.php

```diff
@@ -1263,14 +1263,22 @@ public function getLink()
         $url = $this->getUrl();
 
         if ($url !== null) {
-            $link = '<a href="' . $url . '">' . Html::encode($this->__toString()) . '</a>';
+            $link = '<a href="' . $url . '">' . Html::encode($this->getLinkText()) . '</a>';
 
             return Template::raw($link);
         }
 
         return null;
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function getLinkText()
+    {
+        return $this->__toString();
```

Comment:
Typecasting the element to a string is better than calling `__toString()` directly.
```suggestion
        return (string)$this;
```